File: Sitio Privado/Models/AzureFolder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace Sitio_Privado.Models
{
    public class AzureFolder
    {
        public string Name { get; set; }
        public List<AzureBlob> Blobs { get; set; }
        public List<AzureFolder> Folders { get; set; }

        public AzureFolder(string name)
        {
            Name = name;
            Blobs = new List<AzureBlob>();
            Folders = new List<AzureFolder>();
        }
    }
}

File: Sitio Privado/app/domain/AzureFolder.ts
module app.domain {

    export interface IAzureFolder {
        Name: string;
        Blobs: app.domain.AzureBlob[];
        Folders: app.domain.AzureFolder[];
    }

    export class AzureFolder extends app.domain.EntityBase implements IAzureFolder {
        constructor(public Name: string, public Blobs: app.domain.AzureBlob[], public Folders: app.domain.AzureFolder[]) {

            super();
        }
    }
}